### PR TITLE
Fix bijective mapping in layout.py

### PR DESCRIPTION
**Pull request by rajnisht7**
### Description
This PR fixes bug in layout.py due to which invalid mappings were also accepted instead of raising error
eg.
```
{qr[0]: 0,
qr[1]: 1,
qr[2]: 2}
```
this is fine
but 
```
{qr[0]: 0,
qr[1]: 1,
qr[2]: 1}
```
this type of mapping must raise an error 

which can also be reproduced using
```
from qiskit.transpiler.layout import Layout
from qiskit import QuantumRegister

qr = QuantumRegister(2, 'qr')
test_dict = {
    qr[0]: 1,
    qr[1]: 1
}
layout = Layout(test_dict)     
print("Normal layout created successfully:", layout)
print("Length:", len(layout))
print("p2v:", layout._p2v)
print("v2p:", layout._v2p)
```
and the output is 
```
Normal layout created successfully: Layout({
1: <Qubit register=(2, "qr"), index=1>
})
Length: 1
p2v: {1: <Qubit register=(2, "qr"), index=1>}
v2p: {<Qubit register=(2, "qr"), index=0>: 1, <Qubit register=(2, "qr"), index=1>: 1}
```

### AI/LLM disclosure

- [x] I didn't use LLM tooling, or only used it privately.
- [ ] I used the following tool to help write this PR description:
- [ ] I used the following tool to generate or modify code:

### CODE REVIEW

File: qiskit/transpiler/layout.py [hunk 1]
```diff
@@ -85,12 +85,25 @@ def from_dict(self, input_dict):
...
+            check_physical.add(physical)
+            
         for key, value in input_dict.items():
             virtual, physical = Layout.order_based_on_type(key, value)
             self._p2v[physical] = virtual
-            if virtual is None:
-                continue
-            self._v2p[virtual] = physical
+            if virtual is not None:
+                self._v2p[virtual] = physical
```
Comment (jakelishman) [inline qiskit/transpiler/layout.py:100]: This was unrelated - please try to avoid doing this to minimise code churn.  A maintainer has to review every touched line, so the more there are, the more work it is.

**Review by jakelishman — COMMENTED** [review]
Thanks for the interest in contributing.

This is a complex and comparatively expensive way of doing this.  `_p2v` and `_v2p` are both dictionaries, which already implement set-like semantics over their keys.  For checking `_p2v` it is sufficient to test that the size of the dict matches the size of the input.  For `_v2p` it can be done at insertion time, like
```python
if self._v2p.setdefault(virtual, physical) != physical:
    # ... it's a duplicate
```

Ideally you would have opened a bug report at least coincidentally with the PR, but that's ok.  Bugfixes should come with a suitable test and a release note, though.

File: qiskit/transpiler/layout.py (hunk 1 above)
Comment (rajnisht7) [reply qiskit/transpiler/layout.py:100]: thank you for the feedback, had made the code minimum

**Comment by rajnisht7** [comment]
thank you for the review, have made the changes
-made the code minimal
-added test
-added release note

**Comment by rajnisht7** [comment]
@jakelishman, the implementation is now raising error for duplicate physical qubits and is working fine for edge cases as well.
but in case of duplicate virtual qubits, it is over-writing the previous one
```
{qr[0]: 0, qr[1]: 1, qr[0]: 2} becomes {qr[0]: 2, qr[1]: 1} 
```
which is natural behaviour of dict as well 
but is this expected output or should it raise an error as well

**Comment by rajnisht7** [comment]
@jakelishman Can you kindly take a look on this, whenever you have time. Thank You
